koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -881,65 +853,313 @@ def handler(self, src, target, opts=None):
+                self.logger.error("Failed to merge list of repositories "
+                                  "%r. Reason (%s): %s", repository,
+                                  type(error), error)
+            koji_build_id = result.get('koji_build_id')
+            if koji_build_id:
+                all_koji_builds.append(koji_build_id)
+
+        return {
+            'repositories': all_repositories,
+            'koji_builds': all_koji_builds,
+        }
+
+
+class BuildSourceContainerTask(BaseContainerTask):
+    # Start builds via osbs
+    Methods = ['buildSourceContainer']
+    # Same value as for regular 'build' method.
+    _taskWeight = 2.0
+
+    # JSON Schema definition for koji buildSourceContainer task parameters
+    # Used to validate arguments passed to the handler() method of this class
+    PARAMS_SCHEMA = {
+        "$schema": "http://json-schema.org/draft-07/schema#",
+        "description": "Parameters for a koji buildSourceContainer task.",
+
+        "type": "array",
+        "items": [
+            {
+                "type": "string",
+                "description": "Build target."
+            },
+            {
+                "type": ["object"],
+                "properties": {
+                    "scratch": {
+                        "type": "boolean",
+                        "description": "Perform a scratch build?"
+                    },
+                    "koji_build_id": {
+                        "type": ["integer"],
+                        "description": "Koji build id for sources"
+                    },
+                    "koji_build_nvr": {
+                        "type": ["string"],
+                        "description": "Koji build nvr for sources"
+                    },
+                    "signing_intent": {
+                        "type": ["string"],
+                        "description": "Signing intent of the ODCS composes."
+                    },
+                },
+                "anyOf": [
+                    {"required": ["koji_build_id", "koji_build_nvr"]},
+                    {"required": ["koji_build_nvr"]},
+                    {"required": ["koji_build_id"]}
+                ],
+                "additionalProperties": False
+            }
+        ],
+        "minItems": 2
+    }
+
+    def __init__(self, id, method, params, session, options, workdir=None, demux=False):
+        BaseContainerTask.__init__(self, id, method, params, session, options,
+                                   workdir)
+        self.demux = demux
+
+    def createSourceContainer(self, target_info=None, scratch=None, component=None,
+                              koji_build_id=None, koji_build_nvr=None, signing_intent=None):
+        this_task = self.session.getTaskInfo(self.id)
+        self.logger.debug("This task: %r", this_task)
+        owner_info = self.session.getUser(this_task['owner'])
+        self.logger.debug("Started by %s", owner_info['name'])
+
+        create_build_args = {
```

Comment:
I see that `signing_intent` is missing here, I'll open PR for osbs-client

Reply:
https://github.com/containerbuildsystem/osbs-client/pull/902

Reply:
LGTM